include/grpc/impl/codegen/grpc_types.h

```diff
@@ -615,6 +616,9 @@ typedef struct {
   /** If non-NULL, will be set to point to a string containing the
    * service config used by the channel in JSON form. */
   char** service_config_json;
+  /** If non-NULL, Will be set to the channels connectivity state
+   *  at the time of query. */
+  grpc_connectivity_state state;
```

Comment:
It's not clear to me why C-core should support two different APIs for getting the channel state.  We already have `grpc_channel_check_connectivity_state()`.  Why should we add another?